services/apps/alcs/src/alcs/inquiry/inquiry.service.ts

```diff
@@ -0,0 +1,302 @@
+import { ServiceNotFoundException } from '@app/common/exceptions/base.exception';
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Mapper } from 'automapper-core';
+import { InjectMapper } from 'automapper-nestjs';
+import {
+  FindOptionsRelations,
+  FindOptionsWhere,
+  IsNull,
+  Like,
+  Not,
+  Repository,
+} from 'typeorm';
+import { FileNumberService } from '../../file-number/file-number.service';
+import { formatIncomingDate } from '../../utils/incoming-date.formatter';
+import { filterUndefined } from '../../utils/undefined';
+import { Board } from '../board/board.entity';
+import { CARD_TYPE } from '../card/card-type/card-type.entity';
+import { CardService } from '../card/card.service';
+import { InquiryParcel } from './inquiry-parcel/inquiry-parcel.entity';
+import { InquiryType } from './inquiry-type.entity';
+import {
+  CreateInquiryServiceDto,
+  InquiryDto,
+  UpdateInquiryDto,
+} from './inquiry.dto';
+import { Inquiry } from './inquiry.entity';
+
+@Injectable()
+export class InquiryService {
+  private logger = new Logger(InquiryService.name);
+
+  private CARD_RELATIONS = {
+    board: true,
+    type: true,
+    status: true,
+    assignee: true,
+  };
+
+  private DEFAULT_RELATIONS: FindOptionsRelations<Inquiry> = {
+    card: this.CARD_RELATIONS,
+    localGovernment: true,
+    region: true,
+    type: true,
+  };
+
+  constructor(
+    private cardService: CardService,
+    @InjectRepository(Inquiry)
+    private repository: Repository<Inquiry>,
+    @InjectRepository(InquiryType)
+    private typeRepository: Repository<InquiryType>,
+    @InjectMapper() private mapper: Mapper,
+    private fileNumberService: FileNumberService,
+  ) {}
+
+  async create(createDto: CreateInquiryServiceDto, board?: Board) {
+    await this.fileNumberService.checkValidFileNumber(createDto.fileNumber);
```

Comment:
Are inquiries a manual file number and not auto generated?

Reply:
good catch! done, resolved